src/wp-includes/rest-api.php

```diff
@@ -295,6 +295,25 @@ function rest_api_loaded() {
 	 */
 	define( 'REST_REQUEST', true );
 
+	// Populate the correct $_SERVER variables via an alternate header for fastcgi compatibility.
+	if ( isset( $_SERVER['HTTP_WP_AUTHORIZATION'] ) && preg_match( '%^Basic [a-z\d/+]*={0,2}$%i', $_SERVER['HTTP_WP_AUTHORIZATION'] ) ) {
+		// Removing `Basic ` the token would start six characters in.
+		$token    = substr( $_SERVER['HTTP_WP_AUTHORIZATION'], 6 );
+		$userpass = base64_decode( $token );
+		list( $_SERVER['PHP_AUTH_USER'], $_SERVER['PHP_AUTH_PW'] ) = explode( ':', $userpass );
+	}
+
+	// Determine the user that the request should be run under (if any).
+	if ( isset( $_SERVER['PHP_AUTH_USER'], $_SERVER['PHP_AUTH_PW'] ) ) {
+		// We are explicitly only accepting HTTP Basic Auth for HTTPS requests.
+		if ( ! is_ssl() ) {
+			wp_send_json_error( __( 'HTTP Basic Auth is unavailable for non-HTTPS requests.' ), 403 );
+			die();
+		}
+		$user = wp_authenticate( $_SERVER['PHP_AUTH_USER'], $_SERVER['PHP_AUTH_PW'] );
+		wp_set_current_user( $user->ID );
```

Comment:
This results in a notice when the given username or password is incorrect:

```
Notice: Undefined property: WP_Error::$ID in /var/www/html/wp-includes/rest-api.php on line 314
```

Reply:
Unsure if we want extra clutter for error handling if it fails, but this should address:

```suggestion
		if ( ! is_wp_error( $user ) ) {
			wp_set_current_user( $user->ID );
		}
```

Reply:
added the extra check

Reply:
Did you forget to push this change @draganescu?